app/src/main/java/app/gamenative/ui/component/dialog/ContainerConfigDialog.kt

```diff
@@ -97,8 +98,22 @@ fun ContainerConfigDialog(
             mutableStateOf(initialConfig)
         }
 
-        val screenSizes = stringArrayResource(R.array.screen_size_entries).toList()
-        val graphicsDrivers = stringArrayResource(R.array.graphics_driver_entries).toList()
+    val screenSizes = stringArrayResource(R.array.screen_size_entries).toList()
+    val baseGraphicsDrivers = stringArrayResource(R.array.graphics_driver_entries).toList()
+    var graphicsDrivers by remember { mutableStateOf(baseGraphicsDrivers.toMutableList()) }
+
+        // Perform filesystem discovery off the composition thread to avoid jank
+        LaunchedEffect(Unit) {
+            try {
+                val installed = com.winlator.core.FileUtils.listInstalledAdrenoDrivers(context)
+                if (installed.isNotEmpty()) {
+                    val customDrivers = installed.map { "${it} (Custom)" }
+                    graphicsDrivers = (baseGraphicsDrivers + customDrivers).toMutableList()
+                }
+            } catch (e: Exception) {
+                android.util.Log.w("ContainerConfigDialog", "Failed to discover custom drivers", e)
```

Comment:
I think you'll want Timber here for logging instead

Reply:
I think Utkarsh already did this himself. Driver manager is already in the app.